app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,49 @@
+class QuestionsController < ApplicationController
+  before_action :ensure_logged_in, only:[:new, :create, :update]
+
+  def index
+    @questions = Question.all.order(created_at: :desc)
+  end
+
+  def show
+    @question = Question.includes(:answers, :comments).find(params[:id])
+    @answer = @question.answers.includes(:comments)
+    @fav_answer = @question.answers.where(chosen: true)[0]
+  end
+
+  def new
+    @question = Question.new
+  end
+
+  def create
+    question = current_user.questions.new(question_params)
+    if question.save
+      separate_tags_and_save(params[:tags], question)
+      redirect_to questions_path
+    else
+      @question = Question.new
+      render 'new'
```

Comment:
This will create a new empty question and lose the user's form data. You could do @question = question (or use an ivar in the first place) so that the form keeps the values they entered.